### PR TITLE
Provider - fix idle agreements termination

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**
DO NOT MERGE!
Patch for `beta.1`

### CODE REVIEW

File: agent/provider/src/tasks/task_manager.rs [hunk 1]
```diff
@@ -165,13 +170,23 @@ impl TaskManager {
...
+    fn schedule_idle_expiration(
+        &mut self,
+        msg: ScheduleIdleExpiration,
+        ctx: &mut Context<Self>,
+    ) -> Result<()> {
+        let idle_timeout = msg.0.idle_agreement_timeout;
+        let agreement_id = msg.0.agreement_id;
+
+        // Schedule agreement termination when there is no activity created within timeout.
+        ctx.run_later(idle_timeout.clone(), move |myself, ctx| {
```
Comment (tworec) [inline agent/provider/src/tasks/task_manager.rs:185]: I'm not sure it is enough. What about the case when activity is started and finished between now and now + idle_timeout.

**Review by tworec — CHANGES_REQUESTED** [review]
pls fix commen above

File: agent/provider/src/tasks/task_manager.rs (hunk 1 above)
Comment (nieznanysprawiciel) [reply agent/provider/src/tasks/task_manager.rs]: Thanks for good catch

**Review by tworec — APPROVED** [review]

**Review by tworec — APPROVED** [review]

**Review by mfranciszkiewicz — APPROVED** [review]

**Comment by tworec** [comment]
Please wait for approval before merging this.

**Comment by tworec** [comment]
Tihs is **NOT** to be merged before `beta.1` release. It will be a part of the patch release.
Pls do not merge it before `beta.1` tag is created.

**Review by jiivan — APPROVED** [review]

**Pull request merged**